test/src/unittests/tonal/test_pitchsaliencefunctionpeaks.py

```diff
@@ -0,0 +1,160 @@
+#
+# This file is part of Essentia
+#
+# Essentia is free software: you can redistribute it and/or modify it under
+# the terms of the GNU Affero General Public License as published by the Free
+# Software Foundation (FSF), either version 3 of the License, or (at your
+# option) any later version.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+# FOR A PARTICULAR PURPOSE.  See the GNU General Public License for more
+# details.
+#
+# You should have received a copy of the Affero GNU General Public License
+# version 3 along with this program. If not, see http://www.gnu.org/licenses/
+
+
+from essentia_test import *
+from numpy import *
+
+class TestPitchSalienceFunctionPeaks(TestCase):
+  
+    def testInvalidParam(self):
+        self.assertConfigureFails(PitchSalienceFunctionPeaks(), {'binResolution': -1})
+        self.assertConfigureFails(PitchSalienceFunctionPeaks(), {'maxFrequency': -1})
+        self.assertConfigureFails(PitchSalienceFunctionPeaks(), {'minFrequency': -1})
+        self.assertConfigureFails(PitchSalienceFunctionPeaks(), {'referenceFrequency': -1})          
+
+    def testEmpty(self):
+        self.assertRaises(RuntimeError, lambda: PitchSalienceFunctionPeaks()([]))
+    
+    def testSinglePeak(self):
+        # Provide a single input peak with a unit magnitude at the reference frequency, and
+        # validate that the output salience function has only one non-zero element at the first bin.
+        freq_speaks = [55]
+        mag_speaks = [1]
+        calculatedPitchSalience = PitchSalienceFunction()(freq_speaks,mag_speaks)
+        bins, values = PitchSalienceFunctionPeaks()(calculatedPitchSalience)
+        self.assertEqualVector(bins, [0.])
+        self.assertEqualVector(values, [1.])
+
+    def testDuplicatePeaks(self):
+        # Similar test but for duplicte peaks.
+        # "Mirrors" test3DuplicatePeaks, test case in PitchSalienceFunction.
+        freq_speaks = [55, 55, 55]
+        mag_speaks = [1, 1, 1]
+        calculatedPitchSalience = PitchSalienceFunction()(freq_speaks,mag_speaks)
+        bins, values = PitchSalienceFunctionPeaks()(calculatedPitchSalience)
+        self.assertEqualVector(bins, [0.])
+        self.assertEqualVector(values, [3.])
+
+    def testRegression3Peaks(self):
+        # Provide 3 input peaks with a unit magnitude and
+        # validate that the output salience with previously calculated values
+        freq_speaks = [55, 100, 340]
+        mag_speaks = [1, 1, 1]
+        calculatedPitchSalience = PitchSalienceFunction()(freq_speaks,mag_speaks)
+        bins, values = PitchSalienceFunctionPeaks()(calculatedPitchSalience)
+        self.assertEqualVector(bins, [ 1., 103., 315., 195., 125.,  75.,  37.] )
+        self.assertAlmostEqualVectorFixedPrecision(values, [1.1899976, 1., 1., 0.8, 0.64000005, 0.512, 0.40960002], 6)
```

Comment:
This relative tolerance threshold is too huge. Let's use `assertAlmostEqualVectorFixedPrecision` instead.